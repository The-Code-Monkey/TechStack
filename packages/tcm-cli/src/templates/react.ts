import { PackageJson } from 'type-fest';

import basicTemplate from './basic';
import { Template } from './template';

const reactTemplate: Template = {
  name: 'react',
  dependencies: [
    ...basicTemplate.dependencies,
    '@types/react',
    '@types/react-dom',
    'react',
    'react-dom',
    '@tsconfig/create-react-app',
  ],
  packageJson: {
    ...basicTemplate.packageJson,
    peerDependencies: {
      react: '>=16',
    },
    scripts: {
      ...basicTemplate.packageJson.scripts,
      test: 'tcm test --passWithNoTests',
    } as PackageJson['scripts'],
    jest: {
      testEnvironment: 'jsdom',
    },
  },
};

export default reactTemplate;
