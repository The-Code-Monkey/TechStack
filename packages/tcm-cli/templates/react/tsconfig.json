{
  // see https://www.typescriptlang.org/tsconfig to better understand tsconfigs
  "extends": "../../tsconfig.json",
  "include": ["src", "types"],
"exclude": ["src/**/__test__/*.test.ts*x"],
}
