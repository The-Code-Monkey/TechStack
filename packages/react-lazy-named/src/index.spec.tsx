import { render, screen } from '@testing-library/react';
import * as React from 'react';

import lazy from './index';

const { Suspense } = React;

function Text(props) {
  return props.text;
}

async function fakeImport(result) {
  return { test: result };
}

describe('useWhyChange', () => {
  it('stays quiet if updates never happen', async () => {
    const LazyText = await lazy(() => fakeImport(Text), 'test');

    const { asFragment } = render(
      <Suspense fallback={<Text text="Loading..." />}>
        <LazyText text="hi" />
      </Suspense>
    );

    expect(asFragment()).toMatchSnapshot();

    await screen.findByText('hi');

    expect(asFragment()).toMatchSnapshot();
  });
});
