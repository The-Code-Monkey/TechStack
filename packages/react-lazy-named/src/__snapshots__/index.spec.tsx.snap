// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useWhyChange stays quiet if updates never happen 1`] = `
<DocumentFragment>
  Loading...
</DocumentFragment>
`;

exports[`useWhyChange stays quiet if updates never happen 2`] = `
<DocumentFragment>
  hi
</DocumentFragment>
`;
